code/controllers/subsystem/shuttle.dm

```diff
@@ -282,98 +267,75 @@ SUBSYSTEM_DEF(shuttle)
   * spawned at a generated transit doc. Doing this is how most ships are loaded.
   *
   * * loading_template - The shuttle map template to load. Can NOT be null.
-  * * destination_port - The port the newly loaded shuttle will be sent to after being fully spawned in. Can be null, and will create a transit port if so.
-  * * old_shuttle - The shuttle the newly loaded shuttle will replace(?).
+  * * destination_port - The port the newly loaded shuttle will be sent to after being fully spawned in. If you want to have a transit dock be created, use [proc/load_template] instead. Should NOT be null.
   **/
-/datum/controller/subsystem/shuttle/proc/action_load(datum/map_template/shuttle/loading_template, obj/docking_port/stationary/destination_port = null, obj/docking_port/mobile/old_shuttle = null)
-	// Check for an existing preview
-	if(preview_shuttle && (loading_template != preview_template))
-		preview_shuttle.jumpToNullSpace()
-		preview_shuttle = null
-		preview_template = null
-		QDEL_NULL(preview_mapzone)
-
-	if(!preview_shuttle)
-		if(load_template(loading_template))
-			preview_shuttle.linkup(destination_port)
-		preview_template = loading_template
-
-	// get the existing shuttle information, if any
-	var/timer = 0
-	var/mode = SHUTTLE_IDLE
-	var/obj/docking_port/stationary/D
-
-	if(istype(destination_port))
-		D = destination_port
-	else if(old_shuttle)
-		timer = old_shuttle.timer
-		mode = old_shuttle.mode
-		D = old_shuttle.get_docked()
-
-	if(!D)
-		D = generate_transit_dock(preview_shuttle)
-
-	if(!D)
-		CRASH("No dock found/could be created for preview shuttle ([preview_template.name]), aborting.")
-
-	var/result = preview_shuttle.canDock(D)
-	// truthy value means that it cannot dock for some reason
-	// but we can ignore the someone else docked error because we'll
-	// be moving into their place shortly
-	if((result != SHUTTLE_CAN_DOCK) && (result != SHUTTLE_SOMEONE_ELSE_DOCKED))
-		WARNING("Template shuttle [preview_shuttle] cannot dock at [D] ([result]).")
-		preview_shuttle.jumpToNullSpace()
+/datum/controller/subsystem/shuttle/proc/action_load(datum/map_template/shuttle/loading_template, obj/docking_port/stationary/destination_port)
```

Comment:
Since `destination_port` must no longer be null, it should intentionally crash early.